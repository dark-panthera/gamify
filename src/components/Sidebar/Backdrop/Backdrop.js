import React, { Component } from "react";

const backdrop = props => {
  return <div className="backdrop" onClick={props.onHideBackdrop}></div>;
};

export default backdrop;
