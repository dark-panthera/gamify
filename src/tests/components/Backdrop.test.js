import React from 'react';
import { shallow } from 'enzyme';
import Backdrop from '../../components/Backdrop/Backdrop';

test('should render Backdrop correctly', () => {
  const wrapper = shallow(<Backdrop onHideBackdrop={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

/*
test("should call onHideBackdrop on button click", () => {
    const onHideBackdrop = jest.fn();
    const wrapper = shallow(<Backdrop onHideBackdrop={onHideBackdrop} />);
    wrapper.find('button').simulate('click');
    expect(onHideBackdrop).toHaveBeenCalled();
  });
  */