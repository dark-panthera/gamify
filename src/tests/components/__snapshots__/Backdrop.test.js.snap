// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Backdrop correctly 1`] = `
<div
  className="backdrop"
  onClick={[Function]}
/>
`;
